pxr/imaging/hdSt/materialXFilter.cpp

```diff
@@ -1204,68 +1194,89 @@ _AddMaterialXParams(
         // For filename inputs, manage the associated texture node
         if (varType.getSemantic() == mx::TypeDesc::SEMANTIC_FILENAME) {
             // Get the anonymized MaterialX node name from the param name
-            // annonNodeName_paramName -> annonNodeName
+            // anonNodeName_paramName -> anonNodeName
             std::string mxNodeName = variable->getVariable();
             const auto underscorePos = mxNodeName.find('_');
             if (underscorePos != std::string_view::npos) {
                 mxNodeName = mxNodeName.substr(0, underscorePos);
             }
 
             // Get the original hdNodeName from the MaterialX node name
-            const auto hdNodePathIt = annonToHdNodePathMap.find(mxNodeName);
-            if (hdNodePathIt != annonToHdNodePathMap.end()) {
+            const auto hdNodePathIt = anonToOrigSdfPathMap.find(mxNodeName);
+            if (hdNodePathIt != anonToOrigSdfPathMap.end()) {
                 _UpdateTextureNode(
-                    param.name, hdNetwork, 
+                    param.name, &hdNetwork, 
                     terminalNodePath, hdNodePathIt->second);
             } else {
                 // Storm does not expect textures/filename to be direct inputs 
                 // on materials, replace this filename input with a connection
                 // to an image node
                 _ReplaceFilenameInput(
-                    hdNetwork, terminalNodePath, variable->getVariable());
+                    &hdNetwork, terminalNodePath, variable->getVariable());
             }
         }
     }
 }
 
-static mx::ShaderPtr
-_GenerateMaterialXShader(
-    HdMaterialNetwork2 const& hdNetwork,
+HdSt_MaterialXGeneratorTask::HdSt_MaterialXGeneratorTask(
```

Comment:
This is a different kind of "task" - the necessary data passed to the lambda executing in parallel in TBB and wrapping the MaterialX codegen.